src/which.zig

```diff
@@ -44,8 +52,76 @@ pub fn which(buf: *[bun.MAX_PATH_BYTES]u8, path: []const u8, cwd: []const u8, bi
     return null;
 }
 
+const win_extensionsW = .{
+    bun.strings.w("exe"),
+    bun.strings.w("cmd"),
+    bun.strings.w("bat"),
+};
+const win_extensions = .{ "exe", "cmd", "bat" };
+
+pub fn endsWithExtension(str: []const u8) bool {
+    if (str.len < 4) return false;
+    if (str[str.len - 4] != '.') return false;
+    const file_ext = str[str.len - 3 ..];
+    inline for (win_extensions) |ext| {
+        comptime std.debug.assert(ext.len == 3);
+        if (bun.strings.eqlComptimeCheckLenWithType(u8, file_ext, ext, false)) return true;
+    }
+    return false;
+}
+
+/// This is the windows version of `which`.
+/// It operates on wide strings, and resolves .exe / .cmd extensions.
+/// It is similar to Get-Command in powershell.
+pub fn whichW(buf: *[bun.MAX_PATH_BYTES / 2]u16, path: []const u8, cwd: []const u8, bin: []const u8) ?[:0]const u16 {
+    _ = cwd;
```

Comment:
This should be joined with the `bin` as the first check so that `./node_modules/.bin/prettier` works to become `C:\Users\dave\app\node_modules\.bin\prettier`